From 5caf43a2c36d252bdb48cdebabf4b0badffa7987 Mon Sep 17 00:00:00 2001
From: Ray Strode <rstrode@redhat.com>
Date: Fri, 10 Oct 2014 16:11:01 -0400
Subject: [PATCH 115/119] main: fix incorrectly sized buffer for /proc/cmdline

We assume /proc/cmdline will be no more than 512 bytes (including NUL).
It can actually be 4096 bytes (excluding NUL).

This commit makes sure we allocate enough space for it and its NUL.
---
 src/main.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/main.c b/src/main.c
index dce2f9f..d78ee13 100644
--- a/src/main.c
+++ b/src/main.c
@@ -57,7 +57,7 @@
 #include "ply-progress.h"
 
 #ifndef PLY_MAX_COMMAND_LINE_SIZE
-#define PLY_MAX_COMMAND_LINE_SIZE 512
+#define PLY_MAX_COMMAND_LINE_SIZE 4097
 #endif
 
 #define BOOT_DURATION_FILE     PLYMOUTH_TIME_DIRECTORY "/boot-duration"
@@ -1703,7 +1703,7 @@ get_kernel_command_line (state_t *state)
         }
 
         ply_trace ("reading kernel command line");
-        if (read (fd, state->kernel_command_line, sizeof(state->kernel_command_line)) < 0) {
+        if (read (fd, state->kernel_command_line, sizeof(state->kernel_command_line) - 1) < 0) {
                 ply_trace ("couldn't read it: %m");
                 close (fd);
                 return false;
-- 
1.9.0

